### DISCUSSION TITLE
GitHub Copilot Claude Opus 4.7 pricing not correct

### DISCUSSION THREAD

**Discussion by dehlers-cts** [community/community#192814] [Copilot Conversations] [answered]
### 🏷️ Discussion Type

Product Feedback

### 💬 Feature/Topic Area

Copilot in GitHub

### Body

Claude Opus 4.6 was $5 per million input/$25 per million output tokens. It is charged as 3 premium requests in GitHub Copilot. Now Claude Opus 4.7 is the same $5 per million input/$25 per million output tokens. However, it is being charged at 7.5 premium requests. It seems like GitHub is taking advantage of the new model to make extra money.

**Comment by github-actions** [comment]
**💬 Your Product Feedback Has Been Submitted 🎉**

Thank you for taking the time to share your insights with us! Your feedback is invaluable as we build a better GitHub experience for all our users.

**Here's what you can expect moving forward ⏩**
- Your input will be carefully reviewed and cataloged by members of our product teams. 
  - Due to the high volume of submissions, we may not always be able to provide individual responses.
  - Rest assured, your feedback will help chart our course for product improvements.
- Other users may engage with your post, sharing their own perspectives or experiences. 
- GitHub staff may reach out for further clarification or insight. 
  - We may 'Answer' your discussion if there is a current solution, workaround, or roadmap/changelog post related to the feedback.

**Where to look to see what's shipping 👀**
- Read the [Changelog](https://github.blog/changelog/) for real-time updates on the latest GitHub features, enhancements, and calls for feedback.
- Explore our [Product Roadmap](https://github.com/orgs/github/projects/4247), which details upcoming major releases and initiatives.

**What you can do in the meantime 💻**
- Upvote and comment on other user feedback Discussions that resonate with you.
- Add more information at any point! Useful details include: use cases, relevant labels, desired outcomes, and any accompanying screenshots.

As a member of the GitHub community, your participation is essential. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

Thank you once again for your contribution to making GitHub even better! We're grateful for your ongoing support and collaboration in shaping the future of our platform. ⭐

**Comment by namecallfilter** [comment]
Yeah, and why is it only rolling out to pro+ and not pro too....

**Comment by christianarg** [comment]
Dear github copilot team. Please at least don't remove Opus 4.6. I get it, token cost does not have anything to do with the price you charge for a service. Theory of subjective value by Carl Menger. 
But rolling out Opus4.6 and leaving only us with 4.7 at x7.5 (promotional!?!?) and then at a much higher cost is unreasonable. Leave us the "less powerfull model" for x3 at least. With these multipliers my team will evaluate moving directly to claude code for sure.

**Reply by danilo-css** [reply]
It seems they clearly are trying to make people switch to GPT. Perhaps because Microsoft's original stake on OpenAI gives them cheaper royalties for the models. Who knows.

**Reply by dehlers-cts** [reply, marked as answer]
I found a logical reason why it will cost more. It looks like all coding tools are going to charge more. See below:

Why Opus 4.7 uses more premium requests than Opus 4.6
Two confirmed technical changes directly increase token consumption:

1. Updated tokenizer → 1.0×–1.35× more tokens for the same text
Anthropic states that Opus 4.7’s new tokenizer maps the same input to up to 35% more tokens, depending on content type.

More tokens in → more quota consumed.

2. Higher-effort reasoning → more output tokens
Opus 4.7 “thinks more at higher effort levels,” especially in agentic workflows, which increases output token usage.

This means longer, deeper responses → more quota burned.

**Reply by LTAcosta** [reply]
Having it cost more because of increased premium requests would seem reasonable, but instead we're going to have increased premium requests on top of a multiplier that is over double the multiplier of 4.6? Saying this is a promotional cost implies this is going to get even more expensive? It really does seem like there is a push to get us off Opus here.

**Reply by d4rky-pl** [reply]
@dehlers-cts I wouldn't call this a logical reason considering copilot kneecaps Opus to 160K token window so the higher effort reasoning just gets hit harder. At this point Copilot stops being a steal and starts becoming an increasingly poor value proposition and the multi-model approach is the only thing still keeping it going.

Honestly, if Anthropic didn't decide to burn all the goodwill and ban opencode I would've already made the switch because with current context window limits it's getting near impossible to do any complex work with Copilot alone.

**Reply by PerkyZZ999** [reply]
> I found a logical reason why it will cost more. It looks like all coding tools are going to charge more. See below:
> 
> Why Opus 4.7 uses more premium requests than Opus 4.6
> Two confirmed technical changes directly increase token consumption:
> 
> 1. Updated tokenizer → 1.0×–1.35× more tokens for the same text
> Anthropic states that Opus 4.7’s new tokenizer maps the same input to up to 35% more tokens, depending on content type.
> 
> More tokens in → more quota consumed.
> 
> 2. Higher-effort reasoning → more output tokens
> Opus 4.7 “thinks more at higher effort levels,” especially in agentic workflows, which increases output token usage.
> 
> This means longer, deeper responses → more quota burned.

This shouldnt change anything ... the price is per token so even if there is more or less token the price is still per token so ... the price wont change .. github is just trying to scam people lmao

**Reply by MarkBellaCoreCEO** [reply]
OMG Opus 4.7 15X Yet they move 4.5 and 4.6 to Business only and left Sonnet.  Why force everyone to GPT.  Hell,  I'll just start writing my own LLM and use Cline and LM Studios and then build some vector Titan Mira Brains and to hell with LLM

**Comment by HeDo88TH** [comment]
The 7.5x multiplier is absolutely outrageous, the value has completely disappeared. And it's also supposed to be a promotional multiplier? Are you kidding me?

**Reply by sfacosta** [reply]
Now it's 15x

**Reply by shadowCyber** [reply]
Why is this 15x and not 7.5x?  What happened?

**Comment by Koeta001** [comment]
If they change the pricing like this, the only way is to just go back to Claude code. We have no other choice but Claude code. Then Claude increases the rate limit, then we come back here again, then they increase the price again, till small dev teams can't use AI anymore XD.

**Reply by DockedFerret800** [reply]
I recommend to try Codex. Great rate limits and the quality as well.

**Comment by rodrigoslayertech** [comment]
<img width="807" height="244" alt="image" src="https://github.com/user-attachments/assets/66193a2c-c1ec-4917-b248-ee6c6f65291d" />

LoL?

If this uses 30% more tokens, it should be 4x to be fair, not 7.5x

**Comment by xglukx** [comment]
Dear github copilot team. Please don't be greedy and don't push away the loyal subscribers. It is literally look as scum for now. Even with tokenizer changes for opus 4.7, the price 7.5x for medium is way to high, not telling about 15x after promo ends. I already unsubscribed, and will look for other options, the only one reason to use a copilot was pricing per request and acceptable prices for flagman models of antropic.
Now it will be deefinitelly cheaper just use a claude code with pro plan, and plan the work around their rate limiter, set a sheduler or whatever.
Or time to try codex.
Will take a look on copilot again if something will change regarding the pricing.

**Reply by VertiJoey** [reply]
They communicated in januari this year they were like losing 20 to 80 dollar per month per customer on the 10 an 19 dollar subscribtions, unfortunately things are going change drasticly

**Reply by xglukx** [reply]
Forever used a pro+ and some extra tokens if not enough. Seems they need to review their plans then. It would be fine for higher grade subscription make a discounts, and for lower full price to compensate the losses, just as an idea. But anyway, who will listen us. They do that they want. We just need to adopt.

**Comment by jomardyan** [comment]
Gemini 3.1 Pro seems to be the most capable model. However, in **VS Code** **Chat** it behaves like a complete idiot on agentic tasks. It appears to only reason and generate text, rather than execute actions effectively. Its agentic task capabilities **need significant improvement.** In Antigravity, it works more or less well

**Reply by xglukx** [reply]
Did test the Gemini 3.1, it is feels like far behind the opus in 90% of the task, don't know why. It is circles around, hallucinating, need to repeat multiple times, and do not do what actually were asked. After that just rollback the code, and in 1-2 requests achieve what need to be done with opus. Same agents, same development environment, same skills and tools, but far outcome. Probably for some tasks in is fine, just didn't found common language with Gemini, do not understands me.
Antigravity the agents manager often got hang in the middle of the way with no reason and need to start a new session, and that's incredibly annoying.

Anyway, thanks everyone for comments, will move to Claude code just, should be fine.

**Comment by mehboobme** [comment]
> ### 🏷️ Discussion Type
> Product Feedback
> 
> ### 💬 Feature/Topic Area
> Copilot in GitHub
> 
> ### Body
> Claude Opus 4.6 was $5 per million input/$25 per million output tokens. It is charged as 3 premium requests in GitHub Copilot. Now Claude Opus 4.7 is the same $5 per million input/$25 per million output tokens. However, it is being charged at 7.5 premium requests. It seems like GitHub is taking advantage of the new model to make extra money.

I dont understand why copilot is charging double for 4.7 while anthrpic kept price for opus 4.6 and 4.7 same.

**Comment by joaoalvesmarrucho** [comment]
I've been spending 200€ a months extra-premium requests with Opus 4.6 3X. I dont even want to try 4.7  at 7.5X. Going go back to claude code now.

**Comment by Depot404** [comment]
Opus 4.6 was already expensive — I was spending around $200/month just for that model. Now that you've removed it, the only options left are Opus 4.7 at a 7.5x multiplier, or the older Opus 4.5 at 3x — the exact same cost per request as Opus 4.6, but with a significantly worse model.

I've seen the technical argument floating around: Opus 4.7's new tokenizer maps the same input to up to 35% more tokens, and its higher-effort reasoning produces longer outputs. These are real factors. But let's do the math: 3x × 1.35 = ~4x. Even accounting for increased output tokens from deeper reasoning, we're still nowhere near 7.5x. The gap between what the technical changes justify and what GitHub is actually charging is hard to ignore.

The core problem isn't just the price of Opus 4.7 — it's that Opus 4.6 has been removed entirely, with no like-for-like replacement. Users who were happy paying 3x for a capable model are now forced to either downgrade to Opus 4.5 (older model, same price) or jump to Opus 4.7 at 7.5x — a rate labeled "promotional until April 30" with no disclosure of what comes after.

The Anthropic API pricing for Opus 4.7 is identical to Opus 4.6 ($5/M input, $25/M output), so this isn't purely a cost-driven decision. Removing the mid-tier option while offering no equivalent replacement is a business decision, not a technical necessity.

My ask: keep Opus 4.6 available at 3x, or introduce Opus 4.7 at a comparable multiplier. Forcing users to choose between "worse and cheap" or "better and unaffordable" is not a fair trade-off for a paid Pro+ subscription.

**Reply by IdealisticWolf** [reply]
My Opus 4.6 just disappeared, while using it, from one query to the next, it was gone. Was surprised and had no idea where it went, until I got here.

Cancelled my pro+ plan, going to Claude Code. Only way to reason with these kind of "business decisions" is with your wallet.

**Reply by viniciusviana73** [reply]
@IdealisticWolf Same here. People must leave, so they will not take advantage of us.

**Comment by TimeLordRaps** [comment]
So I cancelled my subscription, however doing research on these topics, I think they are forcing our hand towards the better cheaper models, because for 99% of work they perform the jobs equally or better, example look at gpt 5.4-mini on website design tasks/benchmarks. At this point based on my knowledge of context poisoning over context length, I believe gpt 5.4 at 400k is the worst choice between the top of the line 1x usage models, because of the lost-in-the-middle issues. If anyone has clear definitive benchmarks related to "how many cooperative ideas" an agent can hold in its mind simultaneously, I think its related to number of mha heads[guess], yes codeforce elo of opus4.7 is better, sure but what problems are you really using it on that demand that level of intelligence, 

my guess is that spending 1/7.5x the human time to gain marginal benefits by using a higher quality model is the worse decision then thinking through your own problem with critical thinking and first principles prompt engineering. This means you can't code up your SAAS app in 10 min but instead code up a better engineered more thought out more developed SAAS app in 75min, in human time, this is arbitrary, in compute credits / monetary terms, the difference is the bottom line. 

Plus spending time to think through the problems, makes us smarter, the training data higher quality, and improves the next models faster. Hoping for research in this area to be developed rapidly, but Im busy working on ordinality of super intelligence, so bigger transfinite levels of issues to worry about on my side of things, that I'm no longer choosing to rush through.

I also had to cancel my subscription because Im a recluse no body with no job, so you know a sponsor would help me build safe superintelligence for everybody, just saying.

**Comment by shimonsalamon** [comment]
I get it if 4.7 costs more because it uses more tokens per request — that's a legitimate reason for a higher multiplier. But that doesn't explain removing 4.6.

Anthropic's API pricing is identical for 4.5, 4.6, and 4.7. So keeping 4.6 available at its current 3× costs GitHub nothing. The only reason to remove it is to eliminate the cheaper option and force users onto 4.7 at 7.5×. Leaving Opus 4.5 at 3× as the 'budget' choice doesn't count — it's a clearly worse model, which is why nobody was using it.

I was spending real money on Opus 4.6 at 3× every month and was happy to keep doing so. At 7.5× on 4.7, it's not viable for my workflow. If 4.6 isn't restored, I'll be requesting a refund under the May 20 window and moving to Claude Code, which bills at the actual API rate with no multiplier.

The fix is simple: keep 4.6 in the picker at 3×. Let users choose.

**Comment by JohnnyWalkerDigital** [comment]
I suspect that Opus 4.7's real-world cost is closer to the actual cost to run, and that everything before has been heavily subsidised.

**Comment by ev1ls33d** [comment]


**Reply by SuccessMoneySparkle** [reply]
and they overbill you. they removed 1020 requests from me for 44 question. its nearly 3x overcharging

**Comment by SuccessMoneySparkle** [comment]
Anyone noticed overbilling on premium? I asked 42 question since apr.22. and it says I took 1020 premium requests, when its 7.5 per request it would be 136. so it billed me 3x my usage.

**Comment by profix898** [comment]
The great thing about Copilot was that you could choose your model depending on the complexity of the problem at hand. With the latest changes, it essentially takes that choice away. Anthropic models (esp. Opus) are unreasonably more expensive compared to OpenAI models. I just cancelled my subscription. Still hoping for pricing adjustments in the future though.

**Comment by jomardyan** [comment]
Suprise :D @dehlers-cts @d4rky-pl @LTAcosta @christianarg @profix898 
<img width="452" height="75" alt="image" src="https://github.com/user-attachments/assets/bb8bf9ae-e5cc-4f1d-84a0-64321038be3c" />

**Reply by d4rky-pl** [reply]
<img width="640" height="644" alt="just-wait-a-little-eventually-it-will-get-worse-v0-ze0lysm052v61" src="https://github.com/user-attachments/assets/633a67d6-5cb1-4c01-aa59-95717c69fd7b" />

**Reply by HenningHolm** [reply]
Yep, I feel everything suddenly tok a step back. Now sonnet make me angrier than ever before. Opus 4.7 in my bigger projects can only take like 5 questions per 5 hours. I understand the solution might be to pay more. Still, opus 4.6 made me truly believe AI is taking over, now i dont know. But maybe its for the better... an AI model as good as a serious senior developer might take too much resources... Dont underestimate the human brainpower 🥇

**Comment by SrikanthAttili** [comment]
Anthropic or any frontier AI lab should not release a model at all. if its not commercially viable. There is no point in forcibly keeping us away from opus 4.6, by forcibly rubbing opus 4.7 at double the cost of 4.6. Even if what they said is right about opus 4.7 that its better than 4.6, still we dont want 4.7 if its 2.5 times the cost of 4.6.

**Reply by AngelDimov** [reply]
Did you say double the cost? It's 5x the cost now, and I just cancelled by Copilot subscription.

**Comment by gabrieljolvm** [comment]
It's becoming impossible to use Opus, each request costs about 1.1% of the Pro+, wtf.
<img width="159" height="37" alt="image" src="https://github.com/user-attachments/assets/7824e1ad-b5d9-417e-855c-5dfcc0bc5cd4" />

**Reply by jomardyan** [reply]
> It's becoming impossible to use Opus, each request costs about 1.1% of the Pro+, wtf. <img alt="image" width="159" height="37" src="https://private-user-images.githubusercontent.com/160627642/586376138-7824e1ad-b5d9-417e-855c-5dfcc0bc5cd4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc2MzA2NTAsIm5iZiI6MTc3NzYzMDM1MCwicGF0aCI6Ii8xNjA2Mjc2NDIvNTg2Mzc2MTM4LTc4MjRlMWFkLWI1ZDktNDE3ZS04NTVjLTVkZmNjMGJjNWNkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUwMVQxMDEyMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNWE1NmRhZjU2NWIyNTRhYWY2MjMxYjk2MzdkMTVmYzU5MTNmOTFmYTZhMmVlYzRmMTA1ZjFjY2ZlMDUxMjRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.s5RGYzOL331JWF4NdB3CT30UQm2h42pbv982zwBd0rM">

Purchase and use a direct API if you believe it will be more cost-effective  :D

**Reply by SuccessMoneySparkle** [reply]
they increased it to 15x?

**Reply by MarkBellaCoreCEO** [reply]
Yes

**Comment by MisterQuack** [comment]
15x 
<img width="798" height="893" alt="image" src="https://github.com/user-attachments/assets/52d0458c-b8aa-4464-b8fc-f5b2adbcf167" />

**Comment by CraigLoy** [comment]
Yes I was happily working away using Claude Opus 4.6 at 3x request, as it was much better at interpreting what I want. Now with an overnight change, I'm back to a model that does not understand clearly and really 15x request rate is outrageous for Claude Opus 4.7.

**Comment by JohnnyWalkerDigital** [comment]
I'm not sure the cost is going up. Isn't the multiplier is an indication of how many more tokens a model uses, to help you make a decision on how much of your quota you want to burn. The question is really whether you think 4.7 is is five times better than 4.6 at answering questions -- because it costs 5x as much to run.
